server/src/generated/prisma.graphql

```diff
@@ -1593,6 +1593,7 @@ input TagWhereUniqueInput {
 type User {
   id: ID!
   auth0Id: String
+  key: String
```

Comment:
This is the generated prisma schema, but this changes also appears in our datamodel.
Indeed, we could theorically add a key to any user and they would have access to every query and mutation available to the frontend.

Reply:
That's not exactly what I meant.

Suppose someone has an API key. With the current schema they could call the API to retrieve the list of users with their key. Using those keys they could make new API calls (ask questions etc) impersonating those users. Is that right? Because if it is it looks like a problem.

Reply:
Oh I get what you are saying, and you are right, there is a security risk here. It's slightly more complicated than that, but fortunately there is a simple solution for that!

There is **two** schema in the codebase:
* The **prisma schema** (/server/src/generated/prisma.graphql), generated from the **datamodel** (/server/prisma/datamodel.graphql). In the datamodel, we define all the types used in the application.
* The **backend schema** (/server/schema.graphql). In this schema, we define all Query and Mutations available to our frontend/apis. We also import the basic types (Question, Answer, Tag, User) from our generated prisma schema.

Because we import the User type for our generated prisma schema, it contains every fields (including "key"). And that's where the security risk here.

However, we find a solution here: instead of importing our type "User" in the backend schema, we will redefine a User type with only the needed fields:
```graphql
type User {
  id: ID!
  auth0Id: String

  name: String
  email: String

  picture: String
}
```

This way, the backend **cannot** return a `key` field, as it isn't present in its schema.

What do you think of this solution?

(Commit incoming)

Reply:
That's probably the best solution. :)